Rivet/Functions/Get-RivetConfig.ps1

```diff
@@ -127,9 +127,9 @@ function Get-RivetConfig
         {
             'AsInt'
             {
-                if( -not ($value -is 'int') )
+                if( -not ($value -is [int] -or $value -is [int64]) )
                 {
-                    Write-ValidationError -Message ('setting ''{0}'' is invalid. It should be a positive integer.' -f $Name)
+                    Write-ValidationError -Message ('Setting "{0}" is invalid. It should be a positive integer but we found a "{1}".' -f $Name,$value.GetType().FullName)

```

Comment:
This check does not validate that the integer is positive. Is this an actual requirement? It's probably best that the validation and the messaging match.